### ISSUE TITLE
Accessibility improvements

### ISSUE THREAD

**Issue by isidorn** [Microsoft/vscode#6583]
From @zersiax

I thought it would be something like that. I recommend creating a new issue to report visual indicators on a per-line basis to assistive technology. Ramifications of this would be huge, major accessibility improvement if we pull this off. Proposed flow (feel free to copy to separate issue):

use code for spawning visual indicator next to line for assistive technology logic to determine what lines to flag. Guessing medium difficulty.
Create short, descriptive strings to indicate the function of the indicator. 'has breakpoint' ;has error' 'has warning', 'has monkey'...whatever other indicators exist within Code. I'd be the last person to know which ones exist ;) This isn't all that difficult, but don't make them longer than 2, maybe 3 words.
Now this is where it gets interesting, because there's a number of things we can do here. What comes to mind, from most favorable to least favorable. Please chip in if anyone has other ideas:

Append the string to the end of the line being read by the screenreader. I don't know if this is even possible since you are going to somehow have to track focus within the multiline edit field the code is in and do some ARIA magic to append the string to the end of what is being read, maybe through an invisible live region of some kind.
Play a short sound when a particular indicator is required. A click, a short blip or ding, not entire symphonies please. We are trying to stay productive here :) Again, not sure how technically feasible this is. My JS isn't that advanced yet ;) For this approach the short strings as I suggested above would not be required
Add a hotkey that cycles between the various visual indicators that are present, speak the type of indicator and add a third hotkey to query what for example the error contents are. Perhaps the hotkey for querying error messages is a universal concept to all these ideas, I don't know what the visual flow for this is. D you hover over the indicator and it pops up or is it more involved? The reason this is lower down the list is because we would lose the 'live' aspect. Ideally this is added in after idea 1 or 2 has already been implemented.
Modify the line's formatting and make clear on the accessibility documentation page that screenreaders need to be set to read formatting changes. A lot of users have these settings disabled, and for good reason because it can get quite spammy. Screenreaders can query a line or character's formatting with a hotkey but only do so if they know some useful info might be gleaned from this. This will of course affect sighted users as well though, this is therefore an 'if all else fails' solution.
What are your thoughts on all this? Implementing one of these solutions would fix this issue and a myriad of others, while shooting Code's accessibility up into the stratosphere.

ping @stevencl @alexandrudima

**Comment by zersiax** [comment]
thanks @isidorn

**Comment by isidorn** [comment]
@zersiax unforutentely we did not react on this feedback for over 2 years. I believe the reason for that is that smaller and more concise issues are easier to react to, while reading through this one I feel like it is a collection of multiple.
Are you still using vscode and if yes do you still have similar feedback described in this issues, if yes I would please ask you if you could open smaller atomic issues and ping me on them and we will try to react better.
Since we did not react on this for two years and due to my recommendation above I am closing this one.

Thanks
